src/pages/components/Overlay.qml

```diff
@@ -415,15 +419,23 @@ PanelBackground {
 
             onStatusChanged: {
                 if (activeWebPage && status == PageStatus.Active) {
-                    activeWebPage.grabToFile()
+                    webView.privateMode ? activeWebPage.grabThumbnail() : activeWebPage.grabToFile()
                 }
             }
 
             Browser.TabView {
                 model: webView.tabModel
                 portrait: tabPage.isPortrait
+                privateMode: webView.privateMode
 
                 onHide: pageStack.pop()
+
+                onPrivateModeChanged: {
+                    webView.privateMode = privateMode
+                    tabPage.activeTabIndex =  webView.tabModel.activeTabIndex
+                    tabPage.activeWebPage = webView.contentItem
```

Comment:
Agree.. 

Actually like...
```
                    if (webView.tabModel.count === 0) {
                        overlay.enterNewTabUrl(PageStackAction.Immediate)
                    } else if (!overlayAnimator.atBottom) {
                         dismiss()
                    }
```
I started thinking that with private/non-private modes we could actually stay in tab view when the last tab is closed from tab view regardless of the mode. That would imply that whenever tab count changes to zero (regardless of the mode) and we're at tab we'd execute enterNewTabUrl. When privateMode changes we could only dismiss overlay if tab count more than zero. This approach would be imo more symmetric.

Let's see what @roppola thinks :)